src/woo_publications/utils/query.py

```diff
@@ -0,0 +1,20 @@
+from drf_spectacular.contrib.django_filters import DjangoFilterExtension
+
+from .utils import underscore_to_camel
+
+
+class CamelizeFilterExtension(DjangoFilterExtension):
```

Comment:
this is API related and it's generic/base tooling, so it belongs in the `src/woo_publications/api/` app/package.

It's a good idea to create a `drf_spectacular` package in there, and then create `extensions.py` like in Open Forms: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/api/drf_spectacular/__init__.py

I think you can also just perform the imports in the `__init__.py` modules to register the extension, you don't need to use the `apps.py` for this.